target/linux/realtek/files-5.10/drivers/net/dsa/rtl83xx/rtl930x.c

```diff
@@ -2450,6 +2450,8 @@ static void rtl930x_led_init(struct rtl838x_switch_priv *priv)
 			v = 0x3;
 		if (priv->ports[i].phy_is_integrated)
 			v = 0x1;
+		if (priv->ports[i].is2G5)
+			v = 0x3;
```

Comment:
Why on earth would the number of status LEDs on a port depend on the speed of said port? How many LEDs are available for a specific port is entirely up to the vendor to decide and is independent of de capabilities of a specific port.

Reply:
Because higher speeds are indicated by more leds for example???
Yes, this is heuristics. But it works on all known RTL9300 devices and makes a lot of sense.

Reply:
https://www.netgear.com/business/wired/switches/plus/ms108eup/ has RTL930x, and two LEDs per 2.5G port.

Reply:
As far as I can tell, https://www.zyxel.com/us/en/products_services/12-Port-Web-Managed-Multi-Gigabit-Switch-with-2-Port-2-5G-and-2-Port-10G-SFP--XGS1210-12/ has 2 RGB led's for the two 2.5G ports, so 3 (1x RGB) leds per port. This can be seen on the pcb traces easily as well; 
![image](https://user-images.githubusercontent.com/1245341/181563958-c6c8e34e-761b-4746-b303-b89bc1b595d4.png)
you can see 4 traces, via's and resistors going to the upper and 4 to the lower port led.

I haven't played with the leds in detail yet (will do soon) but from the stock firmware bootup sequence, they show 2 colors per normal led (amber + green) and on the 2.5G they show 'ugly white' and green, to ensure they 'turn on all' leds.

I haven't checked the SFP cages leds, but they blink with two solid colors afaik, blue and something else ...

Reply:
The wiring scheme for the XGS1250-12 is documented on the switch wiki: https://svanheule.net/switches/zyxel_xgs1250-12#port_numbers

It's a weird wiring scheme though. I don't even think you could properly represent all 5 color component in a DT binding.

Reply:
>It's a weird wiring scheme though. I don't even think you could properly represent all 5 color components in a DT description.

They use multiple LEDs on at the same time, like Pink is red plus blue. My 7 year old daughter commented on this, btw and has made a technical request to only have 5GBit connections at home ;-)

I am having the same issue with the GS728TPv2 status LEDs, they are also always double, and e.g. an orange device status because of PoE is created by a green LED plus a red-LED. In the DT I only add the actual LEDs, i.e. red and green.